src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java

```diff
@@ -35,25 +35,29 @@ public OverseasEntitySubmissionDtoValidator(EntityNameValidator entityNameValida
 
     public Errors validateFull(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
 
-        if (isRoeUpdateEnabled && StringUtils.isNotBlank(overseasEntitySubmissionDto.getEntityNumber())) {
-            validateUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
+        if (isRoeUpdateEnabled && isForUpdate(overseasEntitySubmissionDto)) {
+            validateFullUpdateDetails(overseasEntitySubmissionDto, errors, loggingContext);
         } else {
-            validateRegistrationDetails(overseasEntitySubmissionDto, errors, loggingContext);
+            validateFullRegistrationDetails(overseasEntitySubmissionDto, errors, loggingContext);
         }
         return errors;
     }
 
-    private void validateUpdateDetails(OverseasEntitySubmissionDto overseasEntitySubmissionDto, Errors errors, String loggingContext) {
+    private static boolean isForUpdate(OverseasEntitySubmissionDto overseasEntitySubmissionDto) {
```

Comment:
Sorry if not clear, I meant for this to be added as a non-static method to the `OverseasEntitySubmissionDto` class (need to mark method with @JsonIgnore) and then also called from relevant line in `OverseasEntitiesService`.